### PR TITLE
245 wasm wave representation

### PR DESCRIPTION
**Pull request by senia-psm**
Fix: #245 

`--human-readable` flag in `invoke-and-await` is temporary. I'll replace it with a new global format later.

### CODE REVIEW

File: golem-cli/src/worker.rs [hunk 1]
```diff
@@ -79,12 +90,29 @@ pub enum WorkerSubcommand {
         function: String,
 
         /// JSON array representing the parameters to be passed to the function
-        #[arg(short = 'j', long, value_name = "json", value_parser = ValueParser::new(JsonValueParser))]
-        parameters: serde_json::value::Value,
+        #[arg(short = 'j', long, value_name = "json", value_parser = ValueParser::new(JsonValueParser), conflicts_with = "wave")]
+        parameters: Option<serde_json::value::Value>,
+
+        /// Function parameter in wasi-wave format
```
Comment (vigoo) [inline golem-cli/src/worker.rs:96]:
```suggestion
        /// Function parameter in WAVE format
```

Just a minor thing :) the library we use is called `wasm-wave` (not related to wasi) and actually WAVE means "Web Assembly Value Encoding" so i think that's what we should refer to as a format

File: golem-cli/src/model/wave.rs [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+fn wrap_analysed_type(typ: AnalysedType) -> golem_wasm_rpc::AnalysedType {
+    golem_wasm_rpc::AnalysedType(typ)
+}
+
+pub fn wrap_type(typ: &Type) -> golem_wasm_rpc::AnalysedType {
+    wrap_analysed_type(type_to_analysed(typ))
+}
+
+impl WasmFunc for WrapExportFunction {
```
Comment (vigoo) [inline golem-cli/src/model/wave.rs:73]: Ah I did not know there is a function type class in wasm-wave too! That's nice. 
And we should clean this and wasm-rpc up by implementing these typeclasses directly in `golem-wasm-ast` (under a `wave` feature flag). But that can be a separate task of course.

Comment (senia-psm) [reply golem-cli/src/model/wave.rs:73]: I'll move this into `golem-wasm-rpc` after merging this one.
Please note that `DisplayWasmFunc` in `wasm-wave` is not usable for us - it renders `func` as function name (hardcoded).

Comment (vigoo) [reply golem-cli/src/model/wave.rs:73]: I think the correct place for the instances is https://github.com/golemcloud/golem-wasm-ast/blob/main/src/analysis/mod.rs where `AnalysedType` and `AnalysedFunction` is defined.

This means removing the wrapper and instance from `wasm-rpc` too.

**Review by vigoo — APPROVED** [review]

**Review by afsalthaj — APPROVED** [review]

**Pull request merged**